ipywidgets/widgets/widget_selection.py

```diff
@@ -207,11 +207,13 @@ def _propagate_options(self, change):
         self._options_values = tuple(i[1] for i in options)
         if self._initializing_traits_ is not True:
             if len(options) > 0:
-                # if the index is already 0, we set it to None so that setting it to 0
-                # will be recognized as a change (and pick up the new option value and label).
                 if self.index == 0:
-                    self.index = None
-                self.index = 0
+                    # Explicitly trigger the observers to pick up the new value and
+                    # label. Just setting the value would not trigger the observers
+                    # since traitlets thinks the value hasn't changed.
+                    self._notify_trait('index', 0, 0)
```

Comment:
Note that all `_notify_trait` does is calling `notify_change` which is public API.